### DISCUSSION TITLE
Default components props per Theme

### DISCUSSION THREAD

**Discussion by jervalles** [vuetifyjs/vuetify#17879] [General]
I'm using the Theme option by Vuetify for a dark and light.

I saw the new orion patch features to set default props on some components.

Is there a way to set default props for some components but BY theme?

ex: I'm using the "tonal" variant props on kinda every Vbtn in the app, but when I use the Dark theme, I would like to change the variant to these Vbtns (to flat by ex).

**Comment by KaelWD** [comment]
#17192

If you're only changing the theme globally though you can use VDefaultsProvider with a computed ref

**Reply by jervalles** [reply]
You mean that everytime I use a VBtn I have to put a VDefaultsProvider around it?

I don't wanna also put a computed ref on each component that using a Vbtn, I would use a composable. But I will have to import it every time I use a Vbtn also.

**Reply by KaelWD** [reply]
Just one in App.vue
